shared/test_run_query.go

```diff
@@ -210,10 +213,18 @@ func clientSideFilter(
 	to *time.Time,
 	limit *int) (keys []Key, err error) {
 	log := GetLogger(store.Context())
-	log.Debugf("Loading %v viable runs to filter them.", productIDFilter.Cardinality())
-	keys = make([]Key, 0, productIDFilter.Cardinality())
+	capacity := productIDFilter.Cardinality()
+	if productIDFilter.Cardinality() > MaxKeysPerLookup {
+		log.Warningf("%d viable runs exceed the lookup limit %d", productIDFilter.Cardinality(), MaxKeysPerLookup)
+		capacity = MaxKeysPerLookup
+	}
+	log.Debugf("Loading %d viable runs to filter.", capacity)
+	keys = make([]Key, 0, capacity)
 	for key := range productIDFilter.Iter() {
 		keys = append(keys, store.NewIDKey("TestRun", key.(int64)))
+		if len(keys) == capacity {
```

Comment:
Maybe add a note here that runs might be missing?

Reply:
```suggestion
		if len(keys) == capacity {
			// FIXME: This might produce incomplete results.
			// https://github.com/web-platform-tests/wpt.fyi/pull/1914
```

Yeah. Ideally we should tell the users, but short of that we need at least a comment here.